spec/models/equipment_model_spec.rb

```diff
@@ -2,437 +2,343 @@
+        expect { model.not_associated_with_self }.to \
+          change { model.errors }
       end
     end
   end
 
   describe 'counter cache' do
-    before(:each) do
-      @model = FactoryGirl.create(:equipment_model)
-      @res = FactoryGirl.create(:checked_out_reservation,
-                                equipment_model: @model,
-                                start_date: Time.zone.today - 2.days,
-                                due_date: Time.zone.today - 1.day)
-      # have to do this manually because update_overdue marks the res as overdue
-      @res.update_columns(overdue: false)
-      @model.decrement(:overdue_count)
-    end
-    it 'increments when a reservation is marked as overdue' do
-      @res.update_attributes(overdue: true)
-      expect(@model.overdue_count).to eq(1)
-    end
-    it 'only increments once per reservation' do
-      @res.update_attributes(overdue: true)
-      expect { @res.update_attribute(:notes, 'test') }.not_to \
-        change { @model.overdue_count }
-    end
-    it 'decrements when an overdue reservation is checked in' do
-      @res.update_attributes(overdue: true)
-      expect do
-        @res.update_attributes(
-          FactoryGirl.attributes_for(:overdue_returned_reservation,
-                                     equipment_model: @model)
-        )
-      end.to change { @model.overdue_count }.from(1).to(0)
-    end
-    it 'only decrements once per reservation' do
-      @res.update_attributes(overdue: true)
-      @res.update_attributes(
-        FactoryGirl.attributes_for(:overdue_returned_reservation,
-                                   equipment_model: @model)
-      )
-      expect { @res.update_attribute(:notes, 'test') }
-        .not_to change { @model.overdue_count }
-    end
-    it "doesn't change when a normal reservation is checked in" do
-      @res.update_attributes(
-        FactoryGirl.attributes_for(:checked_out_reservation,
-                                   equipment_model: @model)
-      )
-      expect do
-        @res.update_attributes(
-          FactoryGirl.attributes_for(:checked_in_reservation,
-                                     equipment_model: @model)
-        )
-      end.not_to change { @model.overdue_count }
-    end
-    it 'decrements when a reservation is extended' do
-      @res.update_attributes(overdue: true)
-      expect { @res.update_attribute(:due_date, Time.zone.today + 1.day) }.to \
-        change { @model.overdue_count }.from(1).to(0)
+    context 'newly overdue reservation' do
+      it 'increments when a reservation is marked as overdue' do
+        model = FactoryGirl.create(:equipment_model)
+        res = FactoryGirl.create(:checked_out_reservation,
+                                 equipment_model: model,
+                                 start_date: Time.zone.today - 2.days,
+                                 due_date: Time.zone.today - 1.day)
+        res.update_columns(overdue: false)
+        model.decrement(:overdue_count)
```

Comment:
Yes, the callbacks are automatically setting the overdue flag, which causes the cache to be incremented. I'll move these tests to the reservation model spec